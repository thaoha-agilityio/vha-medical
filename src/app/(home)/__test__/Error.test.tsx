import { render } from '@testing-library/react';
import ErrorPage from '../error';

describe('Error page test case', () => {
  const setup = (props: {
    error: Error & { digest?: string };
    reset: () => void;
  }) => render(<ErrorPage {...props} />);

  it('should render correctly', () => {
    const { asFragment } = setup({
      error: new Error('Mock error'),
      reset: jest.fn(),
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
