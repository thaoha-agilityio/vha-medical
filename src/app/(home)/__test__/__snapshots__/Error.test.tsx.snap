// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page test case should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col h-full gap-4 items-center justify-center"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="160"
      src="http://localhost/_next/image?url=%2Fimages%2Ferror.webp&w=384&q=75"
      srcset="/_next/image?url=%2Fimages%2Ferror.webp&w=256&q=75 1x, /_next/image?url=%2Fimages%2Ferror.webp&w=384&q=75 2x"
      style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      width="160"
    />
    <h2
      class="text-foreground text-2xl text-center"
    >
      Oops! There is something wrong
    </h2>
    <p
      class="text-foreground-100 text-lg text-center"
    >
      An error occurred. For more help, feel free to reach out to our support team.
    </p>
    <p
      class="text-foreground-100 text-lg text-center whitespace-pre-line"
    >
      Detail error:
      <br />
      <span
        class="text-danger-100"
      >
        Mock error
      </span>
    </p>
    <p
      class="text-foreground text-lg text-center"
    >
      Please 
      <span
        class="text-secondary-100 hover:underline cursor-pointer"
      >
        reset the page 
      </span>
      or 
      <a
        class="text-secondary-100 hover:underline"
        href="/dashboard"
      >
        back to home
      </a>
    </p>
  </div>
</DocumentFragment>
`;
